src/redux/actionCreators.ts

```diff
@@ -0,0 +1,1474 @@
+import {
+    collides,
+    determineConfigType,
+    getDetails,
+    getExit,
+    getLocalizations,
+    getNode,
+    getNodesBelow,
+    getNodeUI,
+    getPendingConnection,
+    getTranslations,
+    pureSort,
+    snakify
+} from '../utils';
+import {
+    removePendingConnection,
+    updateActionToEdit,
+    updateComponents,
+    updateContactFields,
+    updateCreateNodePosition,
+    updateDefinition,
+    updateDragGroup,
+    updateFetchingFlow,
+    updateFlows,
+    updateFreshestNode,
+    updateGhostNode,
+    updateGroups,
+    updateLanguage,
+    updateLocalizations,
+    updateNodeDragging,
+    updateNodeEditorOpen,
+    updateNodes,
+    updateNodeToEdit,
+    updateOperand,
+    updatePendingConnection,
+    updatePendingConnections,
+    updateResultName,
+    updateResultNames,
+    updateShowResultName,
+    updateTranslating,
+    updateTypeConfig,
+    updateUserAddingAction,
+    updateUserClickingAction,
+    updateUserClickingNode
+} from './actions';
+import {
+    CompletionOption,
+    ComponentDetails,
+    ContactFieldResult,
+    ReduxState,
+    SearchResult
+} from './initialState';
+import { LocalizedObject } from '../services/Localization';
+
+export type DispatchWithState = Dispatch<ReduxState>;
+
+export type GetState = () => ReduxState;
+
+interface Bounds {
+    left: number;
+    top: number;
+    right: number;
+    bottom: number;
+}
+
+interface Reflow {
+    uuid: string;
+    bounds: Bounds;
+}
+
+export interface Connection {
+    previousConnection: Connection;
+}
+
+export interface ConnectionEvent {
+    connection: Connection;
+    source: Element;
+    target: Element;
+    sourceId: string;
+    targetId: string;
+    endpoints: any[];
+}
+
+export type LocalizationUpdates = Array<{ uuid: string; translations?: any }>;
+
+const FORCE_FETCH = true;
+const QUIET_UI = 10;
+const QUIET_SAVE = 1000;
+const NODE_SPACING = 60;
+
+const RESERVED_FIELDS: ContactFieldResult[] = [
+    { id: 'name', name: 'Name', type: 'update_contact' }
+    // { id: "language", name: "Language", type: "update_contact" }
+];
+
+const DEFAULT_OPERAND = '@input';
+
+// let uiTimeout: number;
+let reflowTimeout: number;
+
+export const setTranslating = (translating: boolean) => (
+    dispatch: DispatchWithState,
+    getState: GetState
+) => {
+    if (translating !== getState().translating) {
+        dispatch(updateTranslating(translating));
+    }
+};
+
+export const setLanguage = (language: Language) => (
+    dispatch: DispatchWithState,
+    getState: GetState
+) => {
+    if (!isEqual(language, getState().language)) {
+        dispatch(updateLanguage(language));
+    }
+};
+
+export const setFreshestNode = (freshestNode: Node) => (
+    dispatch: DispatchWithState,
+    getState: GetState
+) => {
+    if (!isEqual(freshestNode, getState().freshestNode)) {
+        dispatch(updateFreshestNode(freshestNode));
+    }
+};
+
+export const setUserClickingAction = (userClickingAction: boolean) => (
```

Comment:
added these 'buffer' methods so we aren't updating state unless we need to. makes for a cleaner  devtools log but not super adamant about this in case there are any objections